### PR TITLE
Utility: Fix retry() exception path to also wait for the retry delay

### LINKED ISSUE #2662: Retry logic is not working properly
```
2021-11-11 03:16:08,885 Error [agora.utils.Utility] - Exception happened while trying to get validator addresses: Bad Gateway
2021-11-11 03:16:09,409 Error [agora.utils.Utility] - Exception happened while trying to get validator addresses: Bad Gateway
2021-11-11 03:16:09,410 Error [agora.utils.Utility] - Exception happened while trying to get validator addresses: Bad Gateway
2021-11-11 03:16:09,410 Error [agora.utils.Utility] - Exception happened while trying to get validator addresses: Bad Gateway
2021-11-11 03:16:09,410 Error [agora.utils.Utility] - Exception happened while trying to get validator addresses: Bad Gateway
2021-11-11 03:16:09,411 Error [agora.utils.Utility] - Exception happened while trying to get validator addresses: Bad Gateway
2021-11-11 03:16:09,411 Error [agora.utils.Utility] - Exception happened while trying to get validator addresses: Bad Gateway
2021-11-11 03:16:09,411 Error [agora.utils.Utility] - Exception happened while trying to get validator addresses: Bad Gateway
2021-11-11 03:16:09,412 Error [agora.utils.Utility] - Exception happened while trying to get validator addresses: Bad Gateway
2021-11-11 03:16:09,412 Error [agora.utils.Utility] - Exception happened while trying to get validator addresses: Bad Gateway
2021-11-11 03:16:09,412 Error [agora.utils.Utility] - Exception happened while trying to get validator addresses: Bad Gateway
2021-11-11 03:16:09,413 Error [agora.utils.Utility] - Exception happened while trying to get validator addresses: Bad Gateway
2021-11-11 03:16:09,413 Error [agora.utils.Utility] - Exception happened while trying to get validator addresses: Bad Gateway
2021-11-11 03:16:09,413 Error [agora.utils.Utility] - Exception happened while trying to get validator addresses: Bad Gateway
2021-11-11 03:16:09,413 Error [agora.utils.Utility] - Exception happened while trying to get validator addresses: Bad Gateway
```

### PR DESCRIPTION
**Pull request by omerfirmak**
Fixes #2662

### CODE REVIEW

**Review by Geod24 — APPROVED** [review]
Yuuuup, thanks!

File: source/agora/utils/Utility.d [hunk 1]
```diff
@@ -59,11 +59,10 @@ Nullable!(ReturnType!dg) retry (alias dg, T)(T waiter, long max_retries,
         {
             if (auto res = dg())
```
Comment (Geod24) [inline source/agora/utils/Utility.d:60]: Although this is dangerous too. We assume that the dg's return value is a true / false thing.
What happens when we call `getBlockHeight` and it returns 0 ?

Comment (omerfirmak) [reply source/agora/utils/Utility.d:60]: Lets make it a `lazy bool`? Using this with anything other than a bool does not really make sense to me anyway

Comment (Geod24) [reply source/agora/utils/Utility.d:60]: Yup, although I got confused with [this](https://github.com/bosagora/agora/blob/b94911fa656bd6ad468bb1feb1a91773df2b204e/source/agora/network/Client.d#L533-L535) so I think we can take care of this another time.

**Pull request merged**